File: api/keys.js
export default function handler(req, res) {
    res.setHeader('Access-Control-Allow-Credentials', true);
    res.setHeader('Access-Control-Allow-Origin', '*');
    res.setHeader('Access-Control-Allow-Methods', 'GET,OPTIONS');
    if (req.method === 'OPTIONS') { res.status(200).end(); return; }

    res.status(200).json({
        GEMINI: process.env.GEMINI_API_KEYS || "",
        SERPER: process.env.SERPER_API_KEYS || "",
        // 📌 ดักจับทุกชื่อเผื่อพิมพ์ผิดใน Vercel รับรองว่าคีย์ไม่หายชัวร์
        FINNHUB: process.env.FINNHUB_API_KEYS || process.env.FINNHUB_KEY_KEYS || process.env.FINNHUB_API_KEY || "",
        ALPHAVANTAGE: process.env.ALPHAVANTAGE_API_KEY || ""
    });
}

File: api/keys.py
import os
import json
from http.server import BaseHTTPRequestHandler

class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        data = {
            "ALPHAVANTAGE": os.environ.get('ALPHAVANTAGE_KEY', ''),
            "SERPER": os.environ.get('SERPER_API_KEYS', ''),
            "GEMINI": os.environ.get('GEMINI_API_KEYS', ''),
            "FINNHUB": os.environ.get('FINNHUB_KEY_KEYS', '')
        }
        
        self.send_response(200)
        self.send_header('Content-type', 'application/json')
        self.send_header('Cache-Control', 'no-store, no-cache, must-revalidate, max-age=0')
        self.send_header('Access-Control-Allow-Origin', '*') 
        self.end_headers()
        
        self.wfile.write(json.dumps(data).encode('utf-8'))
